app/controllers/solutions_controller.rb

```diff
@@ -1,2 +1,31 @@
 class SolutionsController < ApplicationController
+ def create
+    # デバッグ用コメント
+     puts "******************************"
+     puts "********   START    **********"
+     puts "******************************"
+    ans = params[:answer]
+    questionId = params[:question_id]
+    
+    if ans.length > 0
+       puts 'OK'
+      solutions = Solution.new(answer: ans, question_id: questionId)
+      solutions.save
+      flash[:success] = "回答を投稿しました。"
+       redirect_to(question_path)
+      redirect_to("/questions/#{questionId}")
+    else
+       puts 'NG'
+      flash[:danger] = "1つのエラーがあります。"
+       redirect_to(question_path)
+      redirect_to("/questions/#{questionId}")
+    end
+    
+    # デバッグ用コメント
+     puts "******************************"
+     puts "********    END     **********"
+     puts "******************************"
+    
```

Comment:
次のような書き方になります。エラー時にredirectさせると，途中まで書いた文章が全て消えてしまいます。

```suggestion
  def create
    @question = Question.new(question_params)
    if @question.save
      redirect_to questions_path, notice: "質問を投稿しました"
    else
      flash.now[:alert] = "エラーが発生しました"
      @questions = Question.all
      render :index
    end
  end
```